eclipse-scout-core/src/id/Id.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2010, 2024 BSI Business Systems Integration AG
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+import {arrays, Constructor, dates, DoValueMetaData, IdParser, InitModelOf, ObjectModel, objects, ObjectWithType, scout} from '../index';
+
+/**
+ * Abstract representation of an ID.
+ */
+export abstract class Id<TIdValueType, TIdTypeName extends string = string> implements ObjectWithType {
+
+  declare model: IdModel<TIdValueType, TIdTypeName>;
+
+  value: TIdValueType;
+  typeName: TIdTypeName;
+  signature: string;
+  objectType: string;
+
+  init(model: InitModelOf<this>) {
+    const value = scout.assertParameter('value', model?.value);
+    this.typeName = scout.nvl(model.typeName, this.typeName);
+    this.signature = model.signature;
+    this._initIdValue(value);
+  }
+
+  protected abstract _initIdValue(value: TIdValueType | string[]);
+
+  protected _toString(): string {
+    return this.value + '';
+  }
+
+  toUnqualified(): string {
+    const unqualified = this._toString();
+    if (this.signature) {
+      return unqualified + IdParser.SIGNATURE_DELIMITER + this.signature;
+    }
+    return unqualified;
+  }
+
+  toQualified(): string {
+    if (this.typeName) {
+      return this.typeName + IdParser.ID_TYPENAME_DELIMITER + this.toUnqualified();
+    }
+    return this.toUnqualified();
+  }
+
+  static parseTypeName(idMetaData: DoValueMetaData): string {
```

Comment:
No parsing takes place in this function. Is it even called somewhere?

Reply:
Right. I changed the name. Yes, it is used by reflection in IdDoNodeSerializer